xc/xenops_server_xen.ml

```diff
@@ -1012,9 +1008,15 @@ module VM = struct
               for i = 0 to vm.vcpu_max - 1 do
                 Device.Vcpu.add ~xs ~devid:i domid (i < vm.vcpus)
               done;
-              set_domain_type ~xs domid vm
-           );
-      )
+              set_domain_type ~xs domid vm;
+              Some {
+                VmExtra.persistent = persistent;
+                VmExtra.non_persistent = non_persistent
```

Comment:
We're doing quite a lot here under this mutex - do we need to?

Reply:
It's a share that we match on `vmextra` above. We could split it up into a `read` and a smaller `update` though, if you think that is better?

Reply:
It's a bit tricky in this one, because so much happens with `persistent` and `non_persistent`... :(

Reply:
OK!